website/src/pages/api/admin/status.ts

```diff
@@ -0,0 +1,27 @@
+import { getToken } from "next-auth/jwt";
+import { withRole } from "src/lib/auth";
+import { oasstApiClient } from "src/lib/oasst_api_client";
+import { getBackendUserCore } from "src/lib/users";
+
+/**
+ * Returns tasks availability, stats, and tree manager stats.
+ */
+const handler = withRole("admin", async (req, res) => {
+  const token = await getToken({ req });
+  const currentUser = await getBackendUserCore(token.sub);
+  const [tasksAvailabilityData, statsData, treeManagerData] = await Promise.all([
+    oasstApiClient.fetch_tasks_availability(currentUser),
+    oasstApiClient.fetch_stats(),
+    oasstApiClient.fetch_tree_manager(),
+  ]);
```

Comment:
Sorry for the bad code suggestion here. Currently using Promise.all will cause all 3 endpoints either success or error. We should handle the error of them separately. [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#using_promise.allsettled) should be used here.

Reply:
Gotcha, I made the change.